src/include/cache_inline.h

```diff
@@ -445,6 +445,27 @@ __wt_cache_full(WT_SESSION_IMPL *session)
     return (__wt_cache_bytes_inuse(cache) >= conn->cache_size);
 }
 
+/*
+ * __wt_cache_hs_dirty --
+ *     Return if a major portion of the cache is dirty due to history store content.
+ */
+static inline bool
+__wt_cache_hs_dirty(WT_SESSION_IMPL *session)
+{
+    WT_BTREE *hs_btree;
+    WT_CACHE *cache;
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+    cache = conn->cache;
+
+    if (F_ISSET(session, WT_SESSION_NO_DATA_HANDLES) || __wt_hs_get_btree(session, &hs_btree) != 0)
+        return false;
+    return (__wt_cache_bytes_plus_overhead(
+              cache, hs_btree->bytes_dirty_intl + hs_btree->bytes_dirty_leaf) >=
+      ((conn->cache_size / 100) * 20));
```

Comment:
This should be based on the configured dirty cache size, not 20% which reflects the default value.

Reply:
3c7a86c